sdk/eventhub/event-hubs/src/batchingPartitionChannel.ts

```diff
@@ -169,10 +168,23 @@ export class BatchingPartitionChannel {
           ? Math.max(this._maxWaitTimeInMs - timeSinceLastBatchCreation, 0)
           : this._maxWaitTimeInMs;
 
+        const aborter = new AbortController();
+        const abortListener = () => {
+          aborter.abort();
+        };
+        this._loopAbortSignal.addEventListener("abort", abortListener);
+        const updatedOptions = {
+          abortSignal: aborter.signal,
+          abortErrorMsg: StandardAbortMessage
+        };
+        let futureEvent = this._eventQueue.shift(updatedOptions);
+        const delayPromise = delay<void>(maximumTimeToWaitForEvent, updatedOptions.abortSignal, updatedOptions.abortErrorMsg)
         const event =
           eventToAddToBatch ??
-          (await Promise.race([futureEvent, delay<void>(maximumTimeToWaitForEvent)]));
-
+          (await Promise.race([futureEvent, delayPromise]).finally(() => {
```

Comment:
How about this leak-proof race abstraction?
```ts
function race(p1: (signal: AbortSignal) => Promise<T>, p2: (signal: AbortSignal) => Promise<T>, controller: AbortController): Promise<T> {
  return race(p1(controller.signal), p2(controller.signal)).finally(() => {
    aborter.abort();
  })
}
```

Reply:
updated to a similar implementation